In [49]:
#importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [50]:
# loading data
df=pd.read_csv('kc_house_data.csv')
df

id        date     price  bedrooms  bathrooms  sqft_living  \
0      7129300520  10/13/2014  221900.0         3       1.00         1180   
1      6414100192   12/9/2014  538000.0         3       2.25         2570   
2      5631500400   2/25/2015  180000.0         2       1.00          770   
3      2487200875   12/9/2014  604000.0         4       3.00         1960   
4      1954400510   2/18/2015  510000.0         3       2.00         1680   
...           ...         ...       ...       ...        ...          ...   
21592   263000018   5/21/2014  360000.0         3       2.50         1530   
21593  6600060120   2/23/2015  400000.0         4       2.50         2310   
21594  1523300141   6/23/2014  402101.0         2       0.75         1020   
21595   291310100   1/16/2015  400000.0         3       2.50         1600   
21596  1523300157  10/15/2014  325000.0         2       0.75         1020   

       sqft_lot  floors waterfront  view  condition          grade  \
0          5650     1.0        NaN  NONE    Average      7 Average   
1          7242     2.0         NO  NONE    Average      7 Average   
2         10000     1.0         NO  NONE    Average  6 Low Average   
3          5000     1.0         NO  NONE  Very Good      7 Average   
4          8080     1.0         NO  NONE    Average         8 Good   
...         ...     ...        ...   ...        ...            ...   
21592      1131     3.0         NO  NONE    Average         8 Good   
21593      5813     2.0         NO  NONE    Average         8 Good   
21594      1350     2.0         NO  NONE    Average      7 Average   
21595      2388     2.0        NaN  NONE    Average         8 Good   
21596      1076     2.0         NO  NONE    Average      7 Average   

       sqft_above sqft_basement  yr_built  yr_renovated  zipcode      lat  \
0            1180           0.0      1955           0.0    98178  47.5112   
1            2170         400.0      1951        1991.0    98125  47.7210   
2             770           0.0      1933           NaN    98028  47.7379   
3            1050         910.0      1965           0.0    98136  47.5208   
4            1680           0.0      1987           0.0    98074  47.6168   
...           ...           ...       ...           ...      ...      ...   
21592        1530           0.0      2009           0.0    98103  47.6993   
21593        2310           0.0      2014           0.0    98146  47.5107   
21594        1020           0.0      2009           0.0    98144  47.5944   
21595        1600           0.0      2004           0.0    98027  47.5345   
21596        1020           0.0      2008           0.0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
1     -122.319           1690        7639  
2     -122.233           2720        8062  
3     -122.393           1360        5000  
4     -122.045           1800        7503  
...        ...            ...         ...  
21592 -122.346           1530        1509  
21593 -122.362           1830        7200  
21594 -122.299           1020        2007  
21595 -122.069           1410        1287  
21596 -122.299           1020        1357  

[21597 rows x 21 columns]

### Data Understanding

In [51]:
class dataUnderstanding(object):
      
      """ A class that does basic data understanding"""

      def __init__(self, df):
        self.info = df.info
        self.shape = df.shape
        self.describe = df.describe()
        self.duplicates = df.duplicated().sum()
        self.missing = df.isna().sum()
        self.types = df.dtypes

In [52]:
# instantiate the class
understanding = dataUnderstanding(df)

In [53]:
# shape of the dataset
print(f"shape:{understanding.shape}")

shape:(21597, 21)


There are 21597 rows and 21 columns

In [54]:
# info
print(understanding.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21597 entries, 0 to 21596
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21597 non-null  int64  
 1   date           21597 non-null  object 
 2   price          21597 non-null  float64
 3   bedrooms       21597 non-null  int64  
 4   bathrooms      21597 non-null  float64
 5   sqft_living    21597 non-null  int64  
 6   sqft_lot       21597 non-null  int64  
 7   floors         21597 non-null  float64
 8   waterfront     19221 non-null  object 
 9   view           21534 non-null  object 
 10  condition      21597 non-null  object 
 11  grade          21597 non-null  object 
 12  sqft_above     21597 non-null  int64  
 13  sqft_basement  21597 non-null  object 
 14  yr_built       21597 non-null  int64  
 15  yr_renovated   17755 non-null  float64
 16  zipcode        21597 non-null  int64  
 17  lat            21597 non-null  float64
 18  long  

In [55]:
# checking for duplicates
print(f"duplicates;{understanding.duplicates}")

duplicates;0


There are no duplicates

In [56]:
#checking for the missng values
print(understanding.types)

id                 int64
date              object
price            float64
bedrooms           int64
bathrooms        float64
sqft_living        int64
sqft_lot           int64
floors           float64
waterfront        object
view              object
condition         object
grade             object
sqft_above         int64
sqft_basement     object
yr_built           int64
yr_renovated     float64
zipcode            int64
lat              float64
long             float64
sqft_living15      int64
sqft_lot15         int64
dtype: object


In [57]:
understanding.missing

id                  0
date                0
price               0
bedrooms            0
bathrooms           0
sqft_living         0
sqft_lot            0
floors              0
waterfront       2376
view               63
condition           0
grade               0
sqft_above          0
sqft_basement       0
yr_built            0
yr_renovated     3842
zipcode             0
lat                 0
long                0
sqft_living15       0
sqft_lot15          0
dtype: int64

There are 2376 missing values in waterfront column, 63 in the view column and 3842 in the yr_renovated column

### Data Preparation

#### 2.1 Completeness
Taking care of missing values


In [58]:
import missingno as msno
import seaborn as sns
fig, ax = plt.subplots(figsize=(16,8))
msno.bar(df=df, color="blue", ax=ax)
# plt.suptitle("Percentage Completeness")
plt.show()


ImportError: cannot import name 'int' from 'numpy' (c:\anaconda\envs\learn-env\lib\site-packages\numpy\__init__.py)